### ISSUE TITLE
`manifest` Template for Repositories with Multiple `build2` Packages

### ISSUE THREAD

**Issue by lyrahgames** [build2/build2#296]
When packaging a third-party project that contains libraries, tests, examples, and/or executables, we typically use multiple `build2` packages in a single repository to deal with varying dependencies. Each of these packages needs to provide its own `manifest` file that in most cases only slightly differs from the other `manifest` files in the repository. I often see myself copying many information from one `manifest` to another. Also, changing common information requires error-prone copy-and-paste actions for all those files.

I would propose a feature to be able to add a `manifest` template file or a base `manifest` to the packages' repository that may provide all the information that are common between the different package `manifest`s. Then package `manifest` files would only need to add unique information or override wrong values from the base `manifest`.
Further, to not add another file to the whole build system structure, either the `packages.manifest` or the `repositories.manifest` seem to be perfect candidates to provide these common information.

What do you think about such a feature?

**Comment by boris-kolpackov** [comment]
While I understand the motivation, the immediate issue I see with this idea is that `manifest` belongs to the package, not to the repository (unlike `packages.manifest` and `repositories.manifest`, which belong to the repository). To put it another way, when the package directory is distributed into the package archive, the complete `manifest` is expected to be inside, it cannot reference some "base" manifest in the outer directory. Now we could rewrite the "derived" manifest by copying all the relevant data from "base", but this is becoming quite complex both implementation-wise and from the "user confusion" perspective. So I am not sure the benefits of such a functionality warrant the costs.

**Comment by lyrahgames** [comment]
> I was thinking more along the lines of ongoing changes, but, yes, I can see this can also be useful. You could just provide a base manifest and `bdep-new` just copies it to the package manifest. In fact, you could probably already do something like this with the `--post-hook` option in `bdep-new`.

Yes, I am already doing that for some experiments by storing a `manifest` template in the `.develop` folder and make `bdep new --package` copy it as actual `manifest` and inserting all the variables. Still, I didn't use such an approach for packaging a third-party library.
